include/uv/unix.h

```diff
@@ -295,7 +295,8 @@ typedef struct {
   void* queued_fds;                                                           \
   UV_STREAM_PRIVATE_PLATFORM_FIELDS                                           \
 
-#define UV_TCP_PRIVATE_FIELDS /* empty */
+#define UV_TCP_PRIVATE_FIELDS						      \
+  struct sockaddr_storage addr;
```

Comment:
This field breaks ABI compatibility. IIRC, in previous versions you were calling `getsockname()` before calling `sendto()`, that might be a solution to avoid the breakage. Otherwise you should target the PR to the master branch.

Reply:
You probably mean `getpeername()` since the `addr` field is used to store the peer address in the `uv__tcp_connect()` call. However, I don't see a way to set the peer name without actually calling `connect()`.

Is there some other way to add the client-side TCP Fast Open support without breaking the ABI?

Reply:
Target at `master` branch is probably the main way (as then you are permitted to break the API), as there doesn't appear to be unused space elsewhere to fit this (assuming it does need to be stored).

Reply:
:thinking: in that case, is there any rough idea when an ABI change might get into a release? (i.e. current master)  Even a couple years from now might not be too bad, by the looks of this PR's history.